src/test/java/io/jenkins/plugins/analysis/core/model/IssueDifferenceTest.java

```diff
@@ -0,0 +1,63 @@
+package io.jenkins.plugins.analysis.core.model;
+
+import org.junit.jupiter.api.Test;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.LineRange;
+import edu.hm.hafner.analysis.LineRangeList;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+import static java.util.Collections.*;
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests the class {@link IssueDifference}.
+ *
+ * @author Artem Polovyi
+ */
+class IssueDifferenceTest {
+
+    @Test
+    void shouldCreateIssueDifference() {
+
+        Report referenceIssues = new Report().addAll(createFilledIssueWithFingerprint("fixed 1", "fixed fingerprint 1"),
+                createFilledIssueWithFingerprint("fixed 2", "fixed fingerprint 2"),
+                createFilledIssueWithFingerprint("fixed 3", "fixed fingerprint 3"),
+                createFilledIssueWithFingerprint("to fix 1", "to fix fingerprint 1"),
+                createFilledIssueWithFingerprint("to fix 2", "to fix fingerprint 2"));
+
+        Report currentIssues = new Report().addAll(createFilledIssueWithFingerprint("fixed 1", "fixed fingerprint 1"),
+                createFilledIssueWithFingerprint("updated fixed 2", "fixed fingerprint 2"),
+                createFilledIssueWithFingerprint("fixed 3", "fixed fingerprint 3"),
+                createFilledIssueWithFingerprint("new 1", "new fingerprint 1"));
+
+        IssueDifference issueDifference = new IssueDifference(currentIssues, 0, referenceIssues);
```

Comment:
Use a different reference build number to see that everything is correctly working